src/Servers/IIS/IIS/test/IIS.FunctionalTests/Http3Tests.cs

```diff
@@ -0,0 +1,194 @@
+            using var client = new HttpClient(handler);
+            client.DefaultRequestVersion = HttpVersion.Version30;
+            client.DefaultVersionPolicy = HttpVersionPolicy.RequestVersionExact;
+            var ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.GetAsync(address));
+            var qex = Assert.IsType<QuicStreamAbortedException>(ex.InnerException);
+            Assert.Equal(0x010b, qex.ErrorCode);
+        }
+
+        [ConditionalFact]
+        public async Task Http3_ResetAfterHeaders()
+        {
+            var address = Fixture.Client.BaseAddress.ToString() + "Http3_ResetAfterHeaders";
+            var handler = new HttpClientHandler();
+            // Needed on CI, the IIS Express cert we use isn't trusted there.
+            handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+            using var client = new HttpClient(handler);
+            client.DefaultRequestVersion = HttpVersion.Version30;
+            client.DefaultVersionPolicy = HttpVersionPolicy.RequestVersionExact;
+            var response = await client.GetAsync(address, HttpCompletionOption.ResponseHeadersRead);
+            await client.GetAsync(Fixture.Client.BaseAddress.ToString() + "Http3_ResetAfterHeaders_SetResult");
+            response.EnsureSuccessStatusCode();
+            var ex = await Assert.ThrowsAsync<HttpRequestException>(() => response.Content.ReadAsStringAsync());
+            var qex = Assert.IsType<QuicStreamAbortedException>(ex.InnerException?.InnerException?.InnerException);
+            Assert.Equal(0x010c, qex.ErrorCode); // H3_REQUEST_CANCELLED
+        }
+
+        [ConditionalFact]
+        public async Task Http3_AppExceptionAfterHeaders_InternalError()
+        {
+            var address = Fixture.Client.BaseAddress.ToString() + "Http3_AppExceptionAfterHeaders_InternalError";
+            var handler = new HttpClientHandler();
+            // Needed on CI, the IIS Express cert we use isn't trusted there.
+            handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+            using var client = new HttpClient(handler);
+            client.DefaultRequestVersion = HttpVersion.Version30;
+            client.DefaultVersionPolicy = HttpVersionPolicy.RequestVersionExact;
+
+            var response = await client.GetAsync(address, HttpCompletionOption.ResponseHeadersRead);
+            await client.GetAsync(Fixture.Client.BaseAddress.ToString() + "Http3_AppExceptionAfterHeaders_InternalError_SetResult");
+            response.EnsureSuccessStatusCode();
+            var ex = await Assert.ThrowsAsync<HttpRequestException>(() => response.Content.ReadAsStringAsync());
+            var qex = Assert.IsType<QuicStreamAbortedException>(ex.InnerException?.InnerException?.InnerException);
+            Assert.Equal(0x0102, qex.ErrorCode); // H3_INTERNAL_ERROR
```

Comment:
This test fails, the error code is actually `0x02`, shutdown initiated by peer - I believe from here: https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicEnums.cs#L177. This is different from the HTTP/Sys test case - https://github.com/dotnet/aspnetcore/blob/8bcbe51aa2147d8e3a997755f839365e5abe31c8/src/Servers/HttpSys/test/FunctionalTests/Http3Tests.cs#L235. Expected?

Reply:
See
https://github.com/dotnet/aspnetcore/commit/fad96588ab1ed627237b2bdb06a34fcbe876a72a#diff-a78e961f758bf8f4f795410ae852718a6d31dec5599d398f12ffd4c58034d33eR72
https://github.com/dotnet/aspnetcore/blob/8bcbe51aa2147d8e3a997755f839365e5abe31c8/src/Servers/IIS/IIS/src/Core/IISHttpContextOfT.cs#L69